bigflow/dataflow.py

```diff
@@ -1,69 +1,204 @@
 import typing
 import logging
+import uuid
 
 from apache_beam import Pipeline
-from apache_beam.options.pipeline_options import PipelineOptions, GoogleCloudOptions
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.runners.runner import PipelineResult, PipelineState
 
-from bigflow.commons import public
+from typing import Dict, List, Union, Optional, Tuple
+
+try:
+    from typing import TypedDict, Literal
+except ImportError:
+    from typing_extensions import TypedDict, Literal
+
+from bigflow.commons import public, build_docker_image_tag
 from bigflow.workflow import Job, JobContext
-from bigflow.workflow import DEFAULT_EXECUTION_TIMEOUT_IN_SECONDS, DEFAULT_PIPELINE_LEVEL_EXECUTION_TIMEOUT_SHIFT_IN_SECONDS
+
+import bigflow.build.reflect
+
 
 logger = logging.getLogger(__file__)
 
 
+class PipelineOptionsDict(TypedDict, total=False):
```

Comment:
Shouldn't be this public? Also, what is total parameter?

Reply:
It marks all keys as optional, handled by TotalDict metaclass (`typing._TypedDictMeta.__new__`).
https://docs.python.org/3/library/typing.html#typing.TypedDict
> By default, all keys must be present in a TypedDict. It is possible to override this by specifying totality. Usage:

Reply:
👍